website/docs/icons/library/index.js

```diff
@@ -15,15 +15,31 @@ const DEBOUNCE_MS = 250;
 export default class Index extends Component {
   @service router;
 
-  allIcons = catalog.assets.map(({ iconName, fileName, size, description }) => {
-    return {
-      iconName: `${iconName}`,
-      name: `${fileName}`,
-      size: `${size}`,
-      description: `${description}`,
-      searchable: `${iconName}, ${description}`,
-    };
-  });
+  constructor() {
+    super(...arguments);
+    this.groupedIcons = {};
+    // prepare the icons grouped by category
+    this.allIcons.forEach((icon) => {
+      const category = icon.category;
+      if (!this.groupedIcons[category]) {
+        this.groupedIcons[category] = [];
+      }
+      this.groupedIcons[category].push(icon);
+    });
+  }
+
+  allIcons = catalog.assets.map(
```

Comment:
more in general: looking at the logic as is now, you're grouping the icons by category, and then using the "grouped icons" object to perform the seach on it; but in this case you have nested loops which is not very performant (and makes the code more verbose). what if you keep the existing logic to filter the flat array of icons, and only after you have the resulting filtered array you perform the grouping? I think this would streamline the logic and improve the readability of the code (plus you wouldn't need anymore a constructor)

Reply:
[for context: I am aware I told you to look at the tokens code for "inspiration", but there's a subtle difference there: in the tokens logic it needs to also do other "stuff" to the list, so it makes sense to do it once at the beginning; while here the list of icons remains the same]

Reply:
Thanks, I will look at this more closely

Reply:
Technically, is line 67 that should be changed from 
`filteredGroupedIcons = { [icon.category]: [icon] };`
to 
`filteredGroupedIcons[icon.category] = [icon];`

(it would also be consistent withe the other `filteredGroupedIcons` assignments)

Reply:
I made a number of refactors that I pushed up. Let me know what you think or if you have other ideas to improve it

Reply:
Much simpler now, no?